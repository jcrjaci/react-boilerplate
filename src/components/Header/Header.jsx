import React from 'react';
import PropTypes from 'prop-types';
import './Header.scss';

/** Header component
* @param {object} props Component props
* @param {String} props.title Title to display in header.
*/
const Header = ({ title }) => <header className="title">{title}</header>;

Header.propTypes = {
  title: PropTypes.string,
};

Header.defaultProps = {
  title: '',
};

export default Header;
