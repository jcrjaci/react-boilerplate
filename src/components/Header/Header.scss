.title {
  font-size: 2em;
  font-family: sans-serif;
  color: white;
  text-shadow: 0 0 5px black;
  text-align: center;
}