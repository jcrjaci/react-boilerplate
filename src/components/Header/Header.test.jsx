import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Header from './Header';

configure({ adapter: new Adapter() });

describe('<Header />', () => {
  const wrapper = shallow(<Header title="List of top 100 crypto currencies" />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should an empty header', () => {
    wrapper.setProps({ title: '' });
    expect(wrapper.find('.title').text()).toBe('');
  });
});
