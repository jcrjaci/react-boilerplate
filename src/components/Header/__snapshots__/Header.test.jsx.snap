// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<header
  className="title"
>
  List of top 100 crypto currencies
</header>
`;
